test/framework/backuppolicy/backuppolicy.go

```diff
@@ -0,0 +1,182 @@
+		p.Spec.ServiceInstance.Kind = dsi.GetObjectKind().GroupVersionKind().Kind
+		p.Spec.ServiceInstance.Name = dsi.GetName()
+	}
+}
+
+func SetEnabled(enabled bool) Option {
+	return func(p *v1beta3.BackupPolicy) {
+		p.Spec.Enabled = enabled
+	}
+}
+
+func ScheduleConfiguration(scheduleConfig v1beta3.ScheduleConfiguration) Option {
+	return func(p *v1beta3.BackupPolicy) {
+		p.Spec.ScheduleConfig = scheduleConfig
+	}
+}
+
+func RetentionConfiguration(retentionConfig *v1beta3.RetentionConfiguration) Option {
+	return func(p *v1beta3.BackupPolicy) {
+		p.Spec.RetentionConfig = retentionConfig
+	}
+}
+
+func New(opts ...Option) *v1beta3.BackupPolicy {
+	p := &v1beta3.BackupPolicy{}
+	for _, opt := range opts {
+		opt(p)
+	}
+	return p
+}
+
+func GetExistingBackups(p *v1beta3.BackupPolicy, dsi runtimeClient.Object) []v1beta3.Backup {
+	existingBackups := []v1beta3.Backup{
+		{
+			ObjectMeta: v1.ObjectMeta{
+				Name:              framework.UniqueName("backup", suffixLength),
+				Namespace:         p.Namespace,
+				CreationTimestamp: v1.Time{Time: mustParseTime("2024-07-03T00:00:00Z")},
+				Labels: map[string]string{
+					v1beta3.BackupPolicyNameLabelKey: p.Name,
+				},
+			},
+			Spec: v1beta3.BackupSpec{
+				ServiceInstance: p.Spec.ServiceInstance,
+			},
+		},
+		{
+			ObjectMeta: v1.ObjectMeta{
+				Name:              framework.UniqueName("backup", suffixLength),
+				Namespace:         p.Namespace,
+				CreationTimestamp: v1.Time{Time: mustParseTime("2024-07-02T00:00:00Z")},
+				Labels: map[string]string{
+					v1beta3.BackupPolicyNameLabelKey: p.Name,
+				},
+			},
+			Spec: v1beta3.BackupSpec{
+				ServiceInstance: p.Spec.ServiceInstance,
+			},
+		},
+		{
+			ObjectMeta: v1.ObjectMeta{
+				Name:              framework.UniqueName("backup", suffixLength),
+				Namespace:         p.Namespace,
+				CreationTimestamp: v1.Time{Time: mustParseTime("2024-07-01T00:00:00Z")},
+				Labels: map[string]string{
+					v1beta3.BackupPolicyNameLabelKey: p.Name,
+				},
+			},
+			Spec: v1beta3.BackupSpec{
+				ServiceInstance: p.Spec.ServiceInstance,
+			},
+		},
+	}
+	return existingBackups
+}
+
+// CheckBackups waits for the backups matching length.
+func CheckBackups(ctx context.Context, p *v1beta3.BackupPolicy, c runtimeClient.Client, expectedLength int) *v1beta3.BackupList {
+	var err error
+	backups := &v1beta3.BackupList{}
+	EventuallyWithOffset(1, func() bool {
+		lisOptions := []client.ListOption{
+			client.InNamespace(p.Namespace),
+			client.MatchingLabels{v1beta3.BackupPolicyNameLabelKey: p.Name},
+		}
+
+		if err = c.List(ctx, backups, lisOptions...); err != nil {
+			return false
+		}
+
+		if len(backups.Items) == expectedLength {
+			return true
+		}
+		return false
+	}, asyncOpsTimeoutMins, pollingPeriod).Should(BeTrue(),
+		fmt.Sprintf("timeout reached waiting for correct backups count %s/%s: %s",
+			p.GetNamespace(),
+			p.GetName(),
+			err,
+		),
+	)
```

Comment:
I would prefer having the eventually in the test code, instead of in the helper. This makes the test behavior more apparent at first glance.